src/rdkafka_msgset_reader.c

```diff
@@ -868,12 +1032,8 @@ rd_kafka_msgset_reader_v2 (rd_kafka_msgset_reader_t *msetr) {
                 goto done;
         }
 
-        /* Ignore control messages */
-        if (unlikely((hdr.Attributes & RD_KAFKA_MSGSET_V2_ATTR_CONTROL))) {
+        if (hdr.Attributes & RD_KAFKA_MSGSET_V2_ATTR_CONTROL)
```

Comment:
To be safe, move this increment as far up as possible so we don't miss the increment in the case we ignore some ctrl type or whatever.

This counter is used to differentiate between an empty/parse-failed data msgset  or a ctrl msgset, which is then used by the backoff code.